modules/heat_conduction/src/actions/ThermalContactMaterialsAction.C

```diff
@@ -57,7 +57,8 @@ ThermalContactMaterialsAction::act()
 
   InputParameters params = _factory.getValidParams(type);
   // Extract global params
-  _app.parser().extractParams(_name, params);
+  if (isParamValid("parser_syntax"))
+    _app.parser().extractParams(getParam<std::string>("parser_syntax"), params);
```

Comment:
Yes, please get rid of this. Right now only a few places require "parser_syntax", AddICAction in MOOSE, a few places in modules, Bison, and a Yak.